### DISCUSSION TITLE
Functional programming libraries

### DISCUSSION THREAD

**Discussion by brandonjpierce** [gohypergiant/standard-toolkit#12] [General]
Since we use FP for a lot of things I wanted to make sure we assess libraries that exist already that may fit our needs vs. writing everything from scratch. While I don't doubt we could do that, it doesn't really make as much sense to me to take on that much maintenance burden. Very similar to why we would not recreate something like turfjs from scratch.

**Comment by belsrc** [comment]
> Since we use FP for a lot of things
Not as much as I would want. I have an idea on several RFC that I want to spin up regarding control structures and composition. Among some others. 

From the "built or bought", the problem with pretty much all FP libs that I have looked at fall into a couple of different groups. They are either horribly coded from new standards + performance stand point. The other, they are horrible from a flexibility or a standards point of view.

Also, one of the beautiful things about them is that they only have to be done once. Assuming you stick to the laws that dictate them.
